src/test_application/main.cpp

```diff
@@ -0,0 +1,350 @@
+
+    std::string command;
+
+    while(true)
+    {
+        std::cout << "> ";
+        std::cin >> command;
+
+        if (command == "help")
+        {
+            debug_print_help();
+        }
+        else if (command == "simple")
+        {
+            if (destroy_universe())
+            {
+                create_simple_solar_system(g_osp);
+            }
+        }
+        else if (command == "flight")
+        {
+            if (g_magnumThread.joinable())
+            {
+                g_magnumThread.join();
+            }
+            std::thread t(test_flight, std::ref(g_ospMagnum), std::ref(g_osp),
+                          OSPMagnum::Arguments{g_argc, g_argv});
+            g_magnumThread.swap(t);
+        }
+        else if (command == "list_uni")
+        {
+            debug_print_sats();
+        }
+        else if (command == "list_ent")
+        {
+            debug_print_hier();
+        }
+        else if (command == "list_upd")
+        {
+            debug_print_update_order();
+        }
+        else if (command == "exit")
+        {
+            if (g_ospMagnum)
+            {
+                // request exit if application exists
+                g_ospMagnum->exit();
+            }
+            break;
+        }
+        else
+        {
+            std::cout << "that doesn't do anything ._.\n";
+        }
+    }
+
+    // wait for magnum thread to exit if it exists
+    if (g_magnumThread.joinable())
+    {
+        g_magnumThread.join();
+    }
+
+    return 0;
+}
+
+bool destroy_universe()
+{
+    // Make sure no application is open
+    if (g_ospMagnum.get() != nullptr)
+    {
+        std::cout << "Application must be closed to destroy universe.\n";
+        return false;
+    }
+
+    // Destroy all satellites
+    g_osp.get_universe().get_reg().clear();
+
+    // Destroy blueprints as part of destroying all vehicles
+    g_osp.debug_get_packges()[0].clear<BlueprintVehicle>();
+
+    std::cout << "*explosion* Universe destroyed!\n";
+
+    return true;
+}
+
+void load_a_bunch_of_stuff()
+{
+    // Create a new package
+    osp::Package lazyDebugPack("lzdb", "lazy-debug");
+
+    // Load sturdy glTF files
+    osp::AssetImporter::load_sturdy_file("OSPData/adera/spamcan.sturdy.gltf", lazyDebugPack);
+    osp::AssetImporter::load_sturdy_file("OSPData/adera/stomper.sturdy.gltf", lazyDebugPack);
+
+    // Add package to the univere
+    g_osp.debug_get_packges().push_back(std::move(lazyDebugPack));
+
+    // Add 50 vehicles so there's something to load
+    //g_osp.get_universe().get_sats().reserve(64);
+
+    //s_partsLoaded = true;
+
+    std::cout << "Resource loading complete\n\n";
+}
+
+void register_universe_types()
+{
+    osp::universe::Universe &uni = g_osp.get_universe();
+    uni.type_register<osp::universe::SatActiveArea>(uni);
+    uni.type_register<osp::universe::SatVehicle>(uni);
+    uni.type_register<planeta::universe::SatPlanet>(uni);
+}
+
+void debug_print_help()
+{
+    std::cout
```

Comment:
Recommend boost program options, or similar arg parsing library.

Reply:
This part of the code is made to be as simple as possible. I'd like to keep this close to plain c++ to make it easy on new contributes.

Reply:
I mean... Argument parsing libraries make things pretty easy, but its not a big deal either way.